2.2.2 add dist files and bower.json #774

relu opened the pull request.

relu commented:
You might want to create a v2-master branch off of tags/v2.2.1 and eventually merge and tag this there.

Craga89 commented:
Hmm that's weird, I de-registered the package and re-register it at the new qTip2 repo. Doesn't seem to have updated. I'll give that another go

relu commented:
Yes, but that would only cover v3. We'd still be unable to install v2 through bower so these changes are probably still required.

Craga89 commented:
Doh. You're right... my bad, didn't cross my mind! I've re-instated the `qTip2/bower` repo so it should work as before

Craga89 commented:
Thoughts on how to move Bower to the new qTip2 repo without losing the 2.x versions would be appreciated!

relu commented:
My changes are doing just that, check them out (diff on `tags/v2.2.1` to see). I've basically removed `/dist` from `.gitignore`, added the files and changed the version to 2.2.2.
Recommended approach would be for you to create a 2x-master branch (or some similar name) off of v2.2.1 and merge this PR there then tag it v2.2.2.

jskeate commented:
Though that would still be breaking for anyone pointing to a specific version.

Craga89 commented:
Hmm that would only allow people to get the `v2.2.2` package through Bower though no? Anyone trying to get the older versions would get a repo with no dist files...

jskeate commented:
It seems like none of the tags were restored in the bower repo

Craga89 commented:
Interestingly, the reason my `bower unregister qTip2` didn't work is because *bower packages are case sensitive*!? 

These two commands yield different results...

```
bower info qTip2 # The new repo
bower info qtip2  # The old repo
```

Craga89 commented:
@jskeate Fixed that just now

jskeate commented:
Thanks!

relu commented:
@jskeate that is correct, but it's probably a reasonable solution since you'd more easily upgrade from a minor or medium version rather than from a major version.

@Craga89 I suppose you'd need to re-build all the tags and follow a similar approach. Alternatively you'd create a different bower package (maybe qTip3? :))

Craga89 commented:
I wonder if I can use the case-sensitivity to be advantage... re-tagging everything would a monumentous pain!

relu commented:
Not sure if possible and it might cause confusion if it is. I can probably
help you out with re-tagging If you want, might be able to automate the
process.

On Tuesday, 8 December 2015, Craig Michael Thompson <
notifications@github.com> wrote:

> I wonder if I can use the case-sensitivity to be advantage... re-tagging
> everything would a monumentous pain!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qTip2/qTip2/pull/775#issuecomment-163015493>.
>


-- 
Aurel Canciu
+40740208777

Craga89 commented:
Hmm, re-tagging I guess is just a case of branching off from each tag's commit and merging into master + re-tagging once the dist folder is built.

Could probably do it in one big hit so it's not as noticable. Still, feels dirty... and also would still break stuff for people expecting a particular file structure in the bower package.

Hmm...

relu commented:
Here's another dirty approach but at least it would guarantee that things remain pretty much unchanged for the previous versions: create a branch totally unrelated to this repo that would basically mirror the old bower repo and re-create the tags from that. Since you would not need to make any further changes to those, I think this method might actually be pretty safe and straightforward.

Craga89 commented:
Interesting approach... I like it. It hadn't occurred to me that tags span across multiple branches! Great suggestion @relu, hopefully it should be straightforward in that case! Will update this once it's done
